Add a warning to building static libraries with naot (#6651)
dotnet/samples: core/nativeaot/NativeLibrary/README.md

## Building static libraries

> [!WARNING]

> > Building shared libraries is preferred to building static libraries. Static libraries might have problems with ABI differences (the compiler-rt runtime used to build the static libraries that ship with native AOT needs to match the compiler-rt runtime used by other code in the same loadable module). It is also not not possible to mix multiple native AOT compiled static libraries within the same loadable module. These problems don't exist when building a shared library.

Reviewer: ```suggestion
> Building shared libraries is preferred to building static libraries. All code in the loadable module must be compiled with C/C++ compiler version and options that are compatible with native AOT static libraries. It is also not possible to mix multiple native AOT compiled static libraries within the same loadable module. These problems don't exist when building a shared library.
```
`compiler-rt` is one specific example. I would say it in more general way.

> > Building shared libraries is preferred to building static libraries. Static libraries might have problems with ABI differences (the compiler-rt runtime used to build the static libraries that ship with native AOT needs to match the compiler-rt runtime used by other code in the same loadable module). It is also not not possible to mix multiple native AOT compiled static libraries within the same loadable module. These problems don't exist when building a shared library.

Reviewer 2: ```suggestion
> It's preferred to build shared libraries than static libraries:
>
> - Static libraries might have problems with ABI differences&mdash;the compiler-rt runtime used to build the static libraries that ship with Native AOT must match the compiler-rt runtime used by other code in the same loadable module.
> - It's also not possible to mix multiple Native AOT compiled static libraries within the same loadable module.
>
> These problems don't exist when you build a shared library.
```
